src/backend/commands/vacuum.c

```diff
@@ -2003,8 +2003,13 @@ vac_update_datfrozenxid(void)
 	/* chicken out if bogus data found */
 	if (bogus)
 	{
-		/* Cherry-pick from GPDB FIXME: is it right for PG14? */
-		UnLockDatabaseFrozenIds(ExclusiveLock);
+		/*
+		 * Cherry-pick from GPDB FIXME: is it right for PG14?
+		 * Ignore this for error.
+		 */
+#if 0
+		 UnLockDatabaseFrozenIds(ExclusiveLock);
```

Comment:
seem no difference ？

Reply:
It seems that GPDB attempted to resolve a critical bug, but it will encounter issues when transitioning to PG14. To manage this, I’ve implemented a code fix using macros, so that if we encounter the bug again, we can easily address it and save ourselves the hassle of troubleshooting in the future.